onnxruntime/core/providers/cuda/tensor/expand.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "expand.h"
+#include "expand_impl.h"
+#include "core/providers/cpu/tensor/utils.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+#define REGISTER_KERNEL_TYPED(T)                                 \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                 \
+      Expand,                                                    \
+      kOnnxDomain,                                               \
+      8,                                                         \
+      T,                                                         \
+      kCudaExecutionProvider,                                    \
+      KernelDefBuilder()                                         \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()) \
+          .InputMemoryType<OrtMemTypeCPUInput>(1),               \
+      Expand<T>);
+
+void PrintData2(int32_t* buffer, const void* data, int64_t count, std::string name) {
+  cudaMemcpy(buffer, data, count, cudaMemcpyDeviceToHost);
+  std::cout << name << ":" << std::endl;
+  for (int64_t i = 0; i < count; ++i) {
+    std::cout << buffer[i] << std::endl;
+  }
+}
+
+static Status ComputeOutputShape(const std::string& node_name, const TensorShape& lhs_shape, const TensorShape& rhs_shape, TensorShape& out_shape) {
```

Comment:
why static? general we don't use static if don't have to.

Reply:
I would have considered using static to be good. Means there are no dependencies on anything other than the explicit arguments. Easier to test. 

---
In reply to: [297319539](https://github.com/microsoft/onnxruntime/pull/1292#discussion_r297319539) [](ancestors = 297319539)

Reply:
Okay, we can keep it.